lib/vmpooler/api/v1.rb

```diff
@@ -476,6 +476,14 @@ def need_token!
               unless arg.is_a?(Hash)
                 failure = true
               end
+
+              if config['allowed_tags']
+                arg.keys.each do |tag|
+                  unless config['allowed_tags'].include? tag
+                    failure = true
+                  end
```

Comment:
This loop could be avoided by subtracting `arg.keys` from `config['allowed_tags']` and checking for emptiness:

```
>> a = [1, 2, 3, 4, 5]
=> [1, 2, 3, 4, 5]
>> b = [ 2, 3, 4, 5, 6]
=> [2, 3, 4, 5, 6]
>> a - b
=> [1]
>> b - a
=> [6]
```